### PR TITLE
Remove optimization flags from clang codegen tests

### PR DESCRIPTION
**Pull request by MatzeB**
- Remove an -O3 flag from a couple of clang x86 codegen tests so the
  tests do not need to be updated when optimizations in LLVM change.
- Change the tests to use utils/update_cc_test_checks.sh
- Change from apple/darwin triples to generic x86_64-- and
  i386-- because it was not relevant to the test but
  `update_cc_test_checks` seems to be unable to handle platforms that
  prepend `_` to function names.

### CODE REVIEW

**Comment by MatzeB** [comment]
Update clang tests separately as suggested in #113212

File: clang/test/CodeGen/X86/avx-cmp-builtins.c [hunk 1]
```diff
@@ -1,5 +1,6 @@
-// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -O3 -triple=x86_64-apple-darwin -target-feature +avx -emit-llvm -o - | FileCheck %s
-// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -O3 -triple=i386-apple-darwin -target-feature +avx -emit-llvm -o - | FileCheck %s
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 5
+// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -triple=x86_64-- -target-feature +avx -emit-llvm -o - | FileCheck %s --check-prefix=CHECK-X32
+// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -triple=i386-- -target-feature +avx -emit-llvm -o - | FileCheck %s --check-prefix=CHECK-X64
```
Comment (efriedma-quic) [inline clang/test/CodeGen/X86/avx-cmp-builtins.c:3]: Maybe could add `opt -passes=mem2reg` to reduce the amount of clutter in the test output without interacting with "real" optimizations.

File: clang/test/CodeGen/X86/avx-cast-builtins.c [hunk 2]
```diff
@@ -1,101 +1,230 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 5
```
Comment (efriedma-quic) [inline clang/test/CodeGen/X86/avx-cast-builtins.c:1]: I think it makes sense to leave this test as-is... it's serving as an end-to-end test that the given C construct is a no-op in assembly.  If the output changes, that's a bug somewhere.  (We do similar things for other intrinsics we expect to lower to a specific sequence.)

File: clang/test/CodeGen/X86/avx-cmp-builtins.c [hunk 3]
```diff
@@ -1,6 +1,6 @@
-// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -O3 -triple=x86_64-apple-darwin -target-feature +avx -emit-llvm -o - | FileCheck %s
-// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -O3 -triple=i386-apple-darwin -target-feature +avx -emit-llvm -o - | FileCheck %s
-// FIXME: The shufflevector instructions in test_cmpgt_sd are relying on O3 here.
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 5
+// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -triple=x86_64-- -target-feature +avx -disable-O0-optnone -emit-llvm -o - | opt -S -passes=mem2reg | FileCheck %s --check-prefixes=CHECK
+// RUN: %clang_cc1 -flax-vector-conversions=none -ffreestanding %s -triple=i386-- -target-feature +avx -disable-O0-optnone -emit-llvm -o - | opt -S -passes=mem2reg | FileCheck %s --check-prefixes=CHECK
```
Comment (RKSimon) [inline clang/test/CodeGen/X86/avx-cmp-builtins.c:3]: Remove superfluous `--check-prefixes=CHECK`

File: clang/test/CodeGen/X86/avx-shuffle-builtins.c [hunk 4]
```diff
@@ -1,7 +1,7 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 5
 // REQUIRES: x86-registered-target
-// RUN: %clang_cc1 -ffreestanding %s -O3 -triple=x86_64-apple-darwin -target-feature +avx -emit-llvm -o - | FileCheck %s --check-prefixes=CHECK,X64
-// RUN: %clang_cc1 -ffreestanding %s -O3 -triple=i386-apple-darwin -target-feature +avx -emit-llvm -o - | FileCheck %s --check-prefixes=CHECK,X86
-// FIXME: This is testing optimized generation of shuffle instructions and should be fixed.
+// RUN: %clang_cc1 -ffreestanding %s -triple=x86_64-- -target-feature +avx -disable-O0-optnone -emit-llvm -o - | opt -S -passes=mem2reg | FileCheck %s --check-prefixes=CHECK
+// RUN: %clang_cc1 -ffreestanding %s -triple=i386-- -target-feature +avx -disable-O0-optnone -emit-llvm -o - | opt -S -passes=mem2reg | FileCheck %s --check-prefixes=CHECK
```
Comment (RKSimon) [inline clang/test/CodeGen/X86/avx-shuffle-builtins.c:4]: Remove superfluous `--check-prefixes=CHECK`

**Review by RKSimon — APPROVED** [review]
LGTM with minors

**Review by efriedma-quic — APPROVED** [review]
LGTM

**Pull request merged**